src/python/WMCore/Services/RucioConMon/RucioConMon.py

```diff
@@ -0,0 +1,110 @@
+#!/usr/bin/env python
+# coding=utf-8
+"""
+Rucio Consistency Monitor Service class developed on top of the native WMCore
+Service APIs providing custom output and handling as necessary for the
+CMS Workload Management system
+"""
+
+from __future__ import division, print_function, absolute_import
+
+from urllib.parse import urlencode
+
+import json
+import logging
+
+from WMCore.Services.Service import Service
+
+
+class RucioConMon(Service):
+    """
+    API for dealing with retrieving information from Rucio Consistency Monitor
+    """
+
+    def __init__(self, url, logger=None, configDict=None):
+        """
+        Init method for the RucioConMon Class
+        """
+        configDict = configDict or {}
+        configDict.setdefault('endpoint', url)
+        configDict.setdefault('cacheduration', 1)  # in hours
+        configDict.setdefault('accept_type', 'application/json')
+        configDict.setdefault('content_type', 'application/json')
+        configDict['logger'] = logger if logger else logging.getLogger()
+        super(RucioConMon, self).__init__(configDict)
+        self['logger'].debug("Initializing RucioConMon with url: %s", self['endpoint'])
+
+    def _getResult(self, uri, callname="", clearCache=True, args=None):
+        """
+        Either fetch data from the cache file or query the data-service
+        :param uri: The endpoint uri
+        :return:    A dictionary
+        """
+
+        # NOTE: Unlike the common case we are not using the callname for building
+        #       apiUrl. we  are using it only for giving proper name of the cache
+        #       file.
+        # NOTE: The 'callname' should not contain '/', otherwise the cache is
+        #       tried to be created with a a non existing subdirectory structure
+
+        cachedApi = "%s.json" % callname
```

Comment:
Shouldn't this cache name also consider the arguments? Such that a call to `unmerged&rse=SiteA` doesn't hit the same cache as of `unmerged&rse=SiteB`?

Reply:
Same as above.
https://github.com/dmwm/WMCore/blob/eeb8ab2c7083d95a1dc50879d48287f508b33709/src/python/WMCore/Services/CRIC/CRIC.py#L56

Reply:
It's not impossible that they are both wrong. This can probably be verified with unit tests, just to make sure we are not using the wrong cache file. Can you double check it in the next week as you work in the next PRs, Todor?

Reply:
So far the `callname` is used as a unique identifier to the cache name and I am sure we are not overlapping them since I constructed them. For the time being I think this is a good enough.